uiowa/uiowa
# Facilities building API local image construction.

Tracks https://github.com/uiowa/uiowa/issues/6999


## To test
1. checkout branch, ddev composer install, sync the facilities site
2. open `docroot/sites/facilities.uiowa.edu/modules/facilities_core/src/BuildingsProcessor.php`
3. put a breakpoint on line `142`
4. Turn on listening for xDebug
5. run the following command:  
``` 
    ddev xdebug off && ddev drush @facilities.local cr && ddev xdebug on && ddev drush @facilities.local fm-buildings 
```  
6. Observe for a few results that `$result->imageUrl` is being set to the right value, and that a corresponding image is being generated in `docroot/sites/facilities.uiowa.edu/files/building_images`
7. Additionally observe that the alt text for these images is being generated and is useful.
8. if all of that seems good, run the following command:  
```  
    ddev xdebug off && ddev drush @facilities.local fm-buildings 
```  
9. Ensure that all buildings receive an image in the `building_images` folder, and that the process runs smoothly.
10. Test that images for buildings are rendering correctly on their nodes.
11. Check that the images uploaded to the edit page of the nodes are being referenced to the proper location.
12. P R O F I T
